pip/compat/__init__.py

```diff
@@ -25,7 +24,11 @@
 ]
 
 
-uses_pycache = hasattr(imp, 'cache_from_source')
+if sys.version_info < (3, 4):
+    import imp
+    uses_pycache = hasattr(imp, 'cache_from_source')
+else:
+    uses_pycache = True
```

Comment:
> imp.cache_from_source is used in pip.req.req_uninstall.py

Ah, yes, I missed this one !

I feel like the `if (3+)/if(3.4+)/else/else` can probably be made shorter. I'll have a look tomorrow. Thanks.